### DISCUSSION TITLE
프로필 페이지 내 좋아요 기능

### DISCUSSION THREAD

**Discussion by YuHyun-P** [prgrms-fe-devcourse/FEDC2_2JAVATAYO_Dongyoung#246] [Ideas]
#244 
프로필 페이지에서 작성한 글 목록, 좋아요 한 글 목록을 보여주는데 메인 페이지처럼 게시글 상세보기를 하지 않아도 카드의 하트를 누르면 좋아요가 되도록 구현하려 했으나...

![image](https://user-images.githubusercontent.com/96400112/174687407-63a24556-b109-4bd5-8169-33a04fcccaa3.png)

1. 좋아요 버튼 컴포넌트 또한 좋아요 갯수의 동기화 이슈가 있었고, 이를 낙관적 업데이트로 해결함
2. 프로필 페이지 마운트 시에만 카드에 뿌려줄 포스트 정보를 가져옴 → 동기화 이슈🤯
3. 작성한 글 목록, 좋아요 한 글 목록을 Tab UI로 구현하여 탭 메뉴를 스위칭할 때 다시 포스트 정보를 가져오려 했지만, 제가 생각해 본 결과로는 프로필 페이지에서는 useEffect로 탭 스위칭 상태를 감지할 깔끔한 방법이 없음

⇒ 그래서 임의로 프로필 페이지에서는 좋아요를 누를 수 없도록 카드, 좋아요 버튼 컴포넌트에 clickable prop을 추가함

## 🙄 궁금한 점
- [ ] 좋아요를 누를 수 없게 해도 좋은가?!
- [ ] 다른 방법이 있는가?!

**Comment by Hyevvy** [comment]
context에다가 좋아요한 글들의 배열을 저장하고 있지 않나요 ?? 좋아요를 취소하면 이에 대해서 context 에 다시 저장하면 리렌더링 되지 않나욤?
